include/operations/blas_operators.hpp

```diff
@@ -150,44 +151,50 @@ struct syclblas_abs {
+                 : ind2;                                                      \
+    }                                                                         \
+                                                                              \
+    template <typename R1, typename R2>                                       \
+    static R1 eval(R1& r1, size_t ind2, R2& r2) {                             \
+      return (syclblas_abs::eval(r2) op syclblas_abs::eval(r1.getVal()))      \
+                 ? R1(ind2, r2)                                               \
+                 : ((syclblas_abs::eval(r2) ==                                \
+                     syclblas_abs::eval(r1.getVal())) &&                      \
+                    (ind2 op r1.getInd()))                                    \
+                       ? R1(ind2, r2)                                         \
+                       : r1;                                                  \
+    }                                                                         \
+                                                                              \
+    template <typename R>                                                     \
+    static R eval(R r1, R r2) {                                               \
+      return (syclblas_abs::eval(r2.getVal())                                 \
+                  op syclblas_abs::eval(r1.getVal()))                         \
+                 ? r2                                                         \
+                 : ((syclblas_abs::eval(r2.getVal()) ==                       \
+                     syclblas_abs::eval(r1.getVal())) &&                      \
+                    (r2.getInd() op r1.getInd()))                             \
+                       ? r2                                                   \
+                       : r1;                                                  \
+    }                                                                         \
+                                                                              \
+    template <typename R>                                                     \
+    static IndVal<typename R::value_type::value_type> init(const R& r) {      \
+      return IndVal<typename R::value_type::value_type>(                      \
+          0, constant<typename R::value_type::value_type, initial>::value);   \
+    }                                                                         \
+                                                                              \
+    template <typename R1, typename R2>                                       \
+    static IndVal<typename R1::value_type> init(const R1& r1, const R2& r2) { \
+      return IndVal<typename R1::value_type>(                                 \
+          0, constant<typename R1::value_type, initial>::value);              \
+    }                                                                         \
```

Comment:
The whole Ternary operator should be removed and whenever it is used it should be replaced by the Tuple op.